src/PrestaShopBundle/Utils/ImmutableFloat.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace PrestaShopBundle\Utils;
+
+/**
+ * Immutable value-object that contains a float value.
+ */
+class ImmutableFloat
```

Comment:
Originally I started out with `Float`. Then two things crossed my mind:

1. `DateTime` and `DateTimeImmutable`. The immutable one has the word "immutable".
2. Scalar types on PHP >= 7. Consider the following function signature:
    
    ```php
    function doThing(float $bar): Float
    ```
    Namespaces aside, wouldn't it be confusing to have `Float` and `float`?

In addition, from the reader standpoint, if I saw something like this:

```php
$number = Float::fromString($string);
```

The first thing I would think would be "why did this guy implement his own Float type?", and then I would have to reverse-engineer the developer's reasoning to find out why. In my opinion, having the `Immutable` word makes it easier to understand at a glance – it's not about having a java-like Float type that tries to replace the native scalar type, it's about being sure that it cannot be modified and that it will always be valid.

If it was a "real" VO in VO namespace (for example if it was `VO\Amount`), I agree that having the word "Immutable" would be tautological. But since it's so generic, I believe that immutability cannot be taken for granted – you _need_ to read the phpdoc to know.